tests/integration/v3_watch_restore_test.go

```diff
@@ -97,12 +119,23 @@ func TestV3WatchRestoreSnapshotUnsync(t *testing.T) {
 		}
 	}
 
-	// trigger snapshot send from leader to this slow follower
-	// which then calls watchable store Restore
+	// ensure two members has compacted the log twice.
+	if err := logSub.wait(5 * time.Second); err != nil {
+		t.Fatal("Failed to ensure that members compacted Raft log in 5 seconds")
+	}
+	logMonitor.delSubscriber(logSubID)
+	t.Logf("two members have compacted raft logs")
+
+	// After RecoverPartition, leader L will send snapshot to slow F_m0
+	// follower, because F_m0(index:8) is 'out of date' compared to
+	// L(compacted-index:17).
 	clus.Members[0].RecoverPartition(t, clus.Members[1:]...)
-	// We don't expect leadership change here, just recompute the leader'Server index
-	// within clus.Members list.
+	// We don't expect leadership change here, recompute the leader'Server
+	// index within clus.Members list.
 	lead := clus.WaitLeader(t)
+	if lead != initialLead {
+		t.Fatalf("expected leader index (%v), but got (%v)", initialLead, lead)
+	}
```

Comment:
In theory, the leader might change, although usually it will not.  So suggest to remove it.

Reply:
Updated! Please take a look. Thanks!